### PR TITLE
Improve default documentation rendering behaviors 

### PR DESCRIPTION
**Pull request by Josmithr**
"V1" of the library was configured with rendering policies to (mostly) match our existing website rendering. I peeled most of these changes out into a separate PR to avoid conflating the introduction of new infrastructure with rendering improvements. It additionally made before/after comparisons much easier to make.

This PR introduces the rendering improvements I would have otherwise included in the initial package version. It also includes a couple of "fixes" to address issues I discovered later.

Improvements include:
- Add `Alert` rendering for deprecation notices, etc. (this was lacking in the initial implementation as an oversight).
- Add `DEPRECATED` alert column/cells to table containing deprecated items, to make deprecation more immediately evident.
- Change to the default child content ordering of `Package`/`Namespace` as follows: (`Class`, `Function`, `Interface`, `Namespace`, `Variable`, `Type`) -> (`Interface`, `Class`, `Type`, `Function`, `Variable`, `Namespace`).
    - In particular, this hoists `Interface` items above other item kinds, as these are generally the most semantically important.
- Separate static and non-static `Class`/`Interface` members
    - Hoists static members to the top. Makes the usage-wise distinction of different properties, methods, etc. more clear.
- Add `@sealed` and `@virtual` to list of tags added to "Modifiers" table cells by default.
- Add support for `@see` comment blocks (added to the bottom of any parent section as appropriate).
- Add configuration option for specifying text contents to insert into empty table cells.
- Improve header rendering for unnamed/signature items (displays the signature itself in the heading, rather than just something like "(constructor)".

Also update some related documentation.

*Notes:*
- The website's `MarkdownEmitter` implementation has been pre-fixed to correctly handle alert rendering, and fix note-box rendering, but these changes will not be reflected in the site's content until the new package version has been published and adopted, which will occur in a separate PR.

### CODE REVIEW

File: docs/markdown-emitter.js [hunk 1]
```diff
@@ -72,6 +72,25 @@ class HugoMarkdownEmitter extends MarkdownEmitter {
         writer.write(parts[3]); // write trailing whitespace
     }
 
+    /**
+     * Override base logic to make use of Hugo callouts for note boxes
```
Comment (Josmithr) [inline docs/markdown-emitter.js:76]:
```suggestion
     * Override base logic to make use of Hugo callouts for note boxes.
```

File: packages/common/core-interfaces/src/fluidPackage.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export interface IFluidCodeDetails {
 }
 
 /**
- * @deprecated in favor of {@link @fluidframework/container-definitions#isFluidCodeDetails}
+ * @deprecated  in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#isFluidCodeDetails}
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/fluidPackage.ts:103]:
```suggestion
 * @deprecated in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#isFluidCodeDetails}
```

File: packages/common/core-interfaces/src/fluidPackage.ts [hunk 3]
```diff
@@ -111,21 +111,22 @@ export const isFluidCodeDetails = (details: unknown): details is Readonly<IFluid
 };
 
 /**
- * @deprecated in favor of {@link @fluidframework/container-definitions#IFluidCodeDetailsComparer}
+ * @deprecated  in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#IFluidCodeDetailsComparer}
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/fluidPackage.ts:114]:
```suggestion
 * @deprecated in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#IFluidCodeDetailsComparer}
```

File: packages/common/core-interfaces/src/fluidPackage.ts [hunk 4]
```diff
@@ -111,21 +111,22 @@ export const isFluidCodeDetails = (details: unknown): details is Readonly<IFluid
...
+ * {@link @fluidframework/container-definitions/fluidPackage.ts#IProvideFluidCodeDetailsComparer}
  * to have code loading modules in same package.
 */
 export interface IProvideFluidCodeDetailsComparer {
     readonly IFluidCodeDetailsComparer: IFluidCodeDetailsComparer ;
 }
 
 /**
- * @deprecated in favor of {@link @fluidframework/container-definitions#IFluidCodeDetailsComparer}
+ * @deprecated  in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#IFluidCodeDetailsComparer}
```
Comment (Josmithr) [inline packages/common/core-interfaces/src/fluidPackage.ts:129]:
```suggestion
 * @deprecated in favor of {@link @fluidframework/container-definitions/fluidPackage.ts#IFluidCodeDetailsComparer}
```

File: docs/markdown-emitter.js (hunk 1 above)
Comment (Josmithr) [inline docs/markdown-emitter.js:76]:
```suggestion
     * Override base logic to make use of Hugo callouts for note boxes.
```

**Review by tylerbutler — APPROVED** [review]

**Review by tylerbutler — COMMENTED** [review]
🧵

**Pull request merged**